frontend/src/Pages/Favorites.tsx

```diff
@@ -0,0 +1,150 @@
+                        visibility_status: item.visibility_status,
+                        open_content_provider_id: item.open_content_provider_id,
+                        content_url: item.content_url,
+                        channel_title: item.channel_title ?? '',
+                        provider_name: item.provider_name ?? ''
+                    }));
+                setFavorites(mappedFavorites);
+            } catch (err) {
+                setError('Error fetching favorites.');
+                console.error(err);
+            } finally {
+                setIsLoading(false);
+            }
+        };
+        void fetchFavorites();
+    }, [user]);
+
+    const handleUnfavorite = async (
+        contentId: number,
+        type: 'video' | 'library'
+    ) => {
+        try {
+            setIsLoading(true);
+            let endpoint = '';
+
+            if (type === 'video') {
+                endpoint = `videos/${contentId}/favorite`;
+            } else {
+                endpoint = `open-content/${contentId}/save`;
+            }
+
+            const favorite = favorites.find(
+                (fav) => fav.content_id === contentId && fav.type === type
+            );
+
+            if (!favorite) {
+                setError('Favorite not found.');
+                return;
+            }
+
+            const { id, name, open_content_provider_id } = favorite;
+            if (!name || !open_content_provider_id) {
+                setError('Favorite data is incomplete.');
+                return;
+            }
+
+            const payload = {
+                name,
+                content_id: contentId,
+                open_content_provider_id,
+                content_url: `/api/proxy/libraries/${contentId}/`
+            };
+
+            const response = await API.put(endpoint, payload);
+
+            if (response.success) {
+                await mutate('/open-content/favorites');
+                setFavorites((prevFavorites) =>
+                    prevFavorites.filter((fav) => fav.id !== id)
+                );
+
+                if (type === 'library') {
+                    await mutate('/api/libraries');
+                }
+            } else {
+                setError('Failed to unfavorite item.');
+            }
+        } catch (err) {
+            setError('Error unfavoriting item.');
+            console.error(err);
+        } finally {
+            setIsLoading(false);
+        }
+    };
+    if (isLoading) {
+        return <Loading />;
+    }
+
+    if (error && favorites.length !== 0) {
+        return <p className="text-red-500">{error}</p>;
+    }
+
+    return (
+        <div className="px-8 pb-4">
+            <h1>My Favorites</h1>
+            {favorites.length === 0 ? (
+                <p>You have no favorites yet.</p>
+            ) : (
+                <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-6 gap-6 mt-6">
```

Comment:
I am having some odd frontend behavior with the cards, and I think it is probably due to the varying columns. Could you keep it the same as how the library and video cards are with just 4 columns and then we can update it to respond to smaller screens in a different ticket (if needed)
<img width="1512" alt="Screenshot 2024-11-22 at 10 02 04 AM" src="https://github.com/user-attachments/assets/c8147380-be12-4ddd-9e27-4c68fe0c3a39">

Reply:
It also actually has to do with the fixed width of the favorites card-see comment below